homeassistant/components/swidget/__init__.py

```diff
@@ -0,0 +1,71 @@
+"""The swidget integration."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from swidget.discovery import discover_single
+from swidget.exceptions import SwidgetException
+from swidget.swidgetdevice import SwidgetDevice
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .const import DOMAIN
+from .coordinator import SwidgetDataUpdateCoordinator
+
+LOGGER = logging.getLogger(__name__)
+PLATFORMS: list[Platform] = [Platform.LIGHT]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up swidget from a config entry."""
+
+    hass.data.setdefault(DOMAIN, {})
+    try:
+        device = await discover_single(
+            host=entry.data[CONF_HOST],
+            token_name="x-secret-key",
+            password=entry.data[CONF_PASSWORD],
+            use_https=True,
+            use_websockets=True,
+        )
+    except SwidgetException as ex:
+        raise ConfigEntryNotReady(
+            f"Unable to connect to Swidget device to get metadata: {entry.data[CONF_HOST]}"
+        ) from ex
+
+    coordinator = SwidgetDataUpdateCoordinator(hass, device)
+    try:
+        await coordinator.async_config_entry_first_refresh()
+    except Exception as ex:
+        raise ConfigEntryNotReady(
+            f"Unable to to perform initial refresh: {entry.data[CONF_HOST]}"
+        ) from ex
+    hass.data[DOMAIN][entry.entry_id] = coordinator
```

Comment:
```suggestion
    entry.runtime_data = coordinator
```
is the new way and define `type SwidgetConfigEntry = ConfigEntry[SwidgetDataUpdateCoordinator]` and replace all ConfigEntry with `SwidgetConfigEntry`

Reply:
Updated it. That wasn't fun.